src/cluster/detail/ArborX_DistributedDBSCANHelpers.hpp

```diff
@@ -238,11 +236,28 @@ void forwardNeighbors(MPI_Comm comm, ExecutionSpace space, Points const &points,
   Kokkos::resize(space, primitives, num_primitives);
   Kokkos::resize(space, global_boxes, 0); // free space
 
-  BruteForce index(space, primitives);
-  Kokkos::View<int *, MemorySpace> offsets(prefix + "offsets", 0);
-  Kokkos::View<int *, MemorySpace> ranks_to(prefix + "ranks_to", 0);
+  BoundingVolumeHierarchy index(space, primitives);
```

Comment:
Can you share performance numbers?
I expect BF performs better until the number of processes if several thousands.

Reply:
For ~500M points and 3D (~26 neighbor domain boxes), BVH is about 10x faster than BruteForce.